aloha-core/src/test/scala/com/eharmony/aloha/semantics/LazyInitSupport.scala

```diff
@@ -0,0 +1,52 @@
+package com.eharmony.aloha.semantics
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+/**
+  * Provides support for testing that feature functions produced by semantics
+  * do not call the feature function body when optional data is not present.
+  *
+  * This should probably be mixed into unit test companion objects.
+  *
+  * @author deaktator
+  * @since 11/28/2017
+  */
+private[aloha] abstract class LazyInitSupport {
```

Comment:
Making this an abstract class locks down the privacy of the members so they can't be accessed in sub-classes without trickery.